### ISSUE TITLE
Text does not render if Todo list is followed by a Bulleted list that starts with a -

### ISSUE THREAD

**Issue by TheodoreChu** [outline/rich-markdown-editor#321]
Text does not render in the editor if Todo list is followed by a to-do list that starts with a `-`

**To Reproduce**
1. Fork this repository
2. Go to https://github.com/outline/rich-markdown-editor/blob/develop/example/src/index.js
3. Replace example text (line 8) with this: 

const exampleText = \`- [ ] item \n- item\`;

4. Run `yarn start` and open the example at localhost:55555
5. Default text doesn't render
6. Replace the exampleText with this:

const exampleText = \`- item \n- item\`;

7. Start new session to clear cache/localStorage and open the new example. The text renders

**Expected behavior**
The todo list should render with the bulleted list below it

**Version**
 - Version v11.0.4

**Desktop (please complete the following information):**
 - OS: Windows, Ubuntu WSL 18.04 LTS
 - Browser MS Edge, Google Chrome

**Comment by tommoor** [comment]
￼We can see if this can be fixed, but there should be an empty paragraph between lists to be valid, otherwise it's like you changed the list type half way through

**Comment by TheodoreChu** [comment]
I see

> I take it the problematic input was not originally output from the editor?

You're right, the problematic input was not originally output from the editor. I was trying to see if my current notes are compatible with the Rich Markdown Editor when I found the issues.

**Comment by tommoor** [comment]
Yep that seems fair, I think it's unlikely we'll be able to handle the list type changing part way through but the error handling can be improved at a minimum
